### ISSUE TITLE
LineageOS 21: Bitwarden cannot be used for passkey authentication on mobile

### ISSUE THREAD

**Issue by LBJJFCeplwh7X1oUhVIMZqJY1WDDzjnSIsSIJ0o** [bitwarden/mobile#3310]
### Steps To Reproduce

1. Ensure Bitwarden mobile is set as provider for "Passwords, Passkeys and Data Services"
2. Open mobile Browser
3. Open webpage "https://webauthn.io/"
4. Try to register a passkey or authenticate with a passkey

### Expected Result

Android should offer to select the favorite passkey provider to use with the webpage

### Actual Result

Google only offers to store a new or use an already stored passkey in the Google Account or to select another device. The list of other devices only contains NFC, USB or other smartphone.

### Operating System

Android

### Operating System Version

LineageOS 21 (Build: 20240610; Android 14)

### Device

Oneplus 9 Pro (LE2123)

### Build Version

2024.5.1 (10574)

### Beta

- [ ] Using a pre-release version of the application.

**Comment by aphedges** [comment]
I don't think this is a LineageOS-specific issue. I similarly can't create passkeys using a more mainstream setup:

- Operating System: Android
- Operating System Version: Android 14 (Build number: AP2A.240605.024)
- Device: Pixel 6a (Hardware version: MP1.0)
- Build Version: 2024.6.0 (10746)

I tried with the following browsers:

- Chrome (`com.android.chrome`): version 126.0.6478.122
- Firefox (`org.mozilla.firefox`): version 128.0 (Build \#2016030615)

In both browsers, when I tried to create a passkey for <https://webauthn.io/>, Google's password manager popped up instead as the only option for storage. Additionally, when I created an account for <https://webauthn.io/> on my desktop, I was able to use that account to log in for both Chrome and Firefox.

I was unable to use Bitwarden when I attempted to log into an existing account on a website, but I think it is related to the following limitation (from [Setup Bitwarden for use with passkeys](https://bitwarden.com/help/auto-fill-android/#setup-bitwarden-for-use-with-passkeys)):

> Please also note that Android does not allow 3rd party passkey providers like Bitwarden to support passkey-based 2FA (a.k.a. "non-discoverable credentials").

**Comment by 9SMTM6** [comment]
I've just submitted a suggestion to that article.

Perhaps youve got the same issue.

On my CMF Phone I've got the additional field 'Additional Providers' in the relevant android settings.

When I unselected the only option, google, there, it did not offer to save passkeys in google any longer. But any Passkey registration simply failed.

I had to switch the regular option to none and then back to bitwarden (after unselecting google in the additional option of course) to make bitwarden offer to save passkeys when registering them.

**Comment by somova** [comment]
Are there any news in this topic?

**Comment by msiqx9** [comment]
I was able to get [passkeys](https://github.com/bitwarden/mobile/issues/3253#issuecomment-2324304552) to work in LineageOS 21.

**Comment by msiqx9** [comment]
I dug out my old OnePlus 5 where I had previously wiped the OS and installed LineageOS 21 (based on Android 14). This phone has Bitwarden app 2024.8.0 (beta) and Chrome browser v127 installed and set "Enabled for Google Password Manager and 3rd party passkeys".

I had success with WebAuth.io with the passkey being stored and used from Bitwarden.

**Comment by braupach** [comment]
I could reproduce this bug on my Poco F1 but it now works after the 21.0-20240910 update. 
I guess this issue can be closed now?

**Comment by trandzik** [comment]
Please will it be possible to use Bitwarden for passkeys without Google Play Services on LineageOS?

When trying to sign in with a passkey on following website https://www.passkeys.io/ after clicking on the button nothing happens.

I am using Bitwarden 2024.7.1, Firefox 130.0.1, LineageOS 21.

**Comment by msiqx9** [comment]
I have just updated my OnePlus 5 to 21-20240930-NIGHTLY-cheeseburger and am running the Bitwarden app 2024.8.1 (beta).  I have just tested [https://webauthn.io](url) in the Chrome browser v129 and have had success with the passkey being stored in and used from Bitwarden.

**Comment by somova** [comment]
Unfortunately, the problem still persists. As a result, the Bitwarden app is unfortunately only partially usable on Android. It now seems to work in the Chrome browser. However, there are other cases in which it does not work (for example, registration in the Vivaldi browser).

**Comment by msiqx9** [comment]
I updated my OnePlus 5 to 22.1-20241231-NIGHTLY-cheeseburger (Android 15) with Bitwarden app 2024.12.0 and Chrome browser 131 with Android Credential Management enable set to 3rd party passkeys only.  

Now passkeys stored/accessed in Bitwarden work with webauthn.io and crowdin.com but I have seen some occasional failures.

**Comment by somova** [comment]
The latest version (7.2.3628.68) of Vivaldi for Android (com.vivaldi.browser) still does not work with the Bitwarden app. It is still not possible, to register new passkeys using webauthn.io. Does anybody has similar experiences?

**Comment by closebot-bw** [comment]
This repo has been archived. To report an issue you're experiencing in the new native apps, visit the [iOS](https://github.com/bitwarden/ios) / [Android](https://github.com/bitwarden/android) repo.
